qcodes/instrument_drivers/oxford/fridge_http_server.py

```diff
@@ -0,0 +1,231 @@
+            temp['unit'] = parameter.unit
+            triton_parameters.append(temp)
+        hostname = socket.gethostname()
+        state = {"ts": time.time(), "parameters": triton_parameters, 'hostname': hostname}
+
+        return state
+
+    def run_app(self, loop):
+        app = web.Application()
+        app.router.add_get('/', self.index)
+
+        # construct a regex matching all parameters that the
+        # triton driver exposes.
+        parameter_regex = ""
+        settable_parameter_regex = ""
+        for parameter in self.triton.parameters:
+            parameter_regex += parameter
+            parameter_regex += "|"
+            if getattr(self.triton, parameter).has_set:
+                settable_parameter_regex += parameter
+                settable_parameter_regex += '|'
+        parameter_regex = parameter_regex[0:-1]
+        settable_parameter_regex = parameter_regex[0:-1]
+
+        # route all parameters to handle_parameter.
+        # The slightly cryptic syntax below means {{parametername:paramregex}}.
+        # means route all matching the paramregex and make the parameter known
+        # to the handler as parametername. {{ is needed for literal { in format strings
+        app.router.add_get('/{{parametername:{}}}'.format(parameter_regex), self.handle_parameter)
+        app.router.add_post('/{{parametername:{}}}'.format(settable_parameter_regex), self.handle_parameter)
+        app.router.add_get('/', self.index)
+        app.router.add_get('/hostname', self.handle_hostname)
+        return app
+
+    async def websocket_client(self):
+        ws_addr = qcodes.config['fridgeserver']['wsserver']
+        ws_port = qcodes.config['fridgeserver']['wsport']
+        ws_server = '{}:{}'.format(ws_addr, ws_port)
+        ws_sleep = qcodes.config['fridgeserver']['wssendinterval']
+        ws_connect_sleep = qcodes.config['fridgeserver']['wsconnectinterval']
+        while True:
+            try:
+                log.debug("Connecting via websockets to {}".format(ws_server))
+                async with websockets.connect(ws_server) as websocket:
+                    log.debug("connected from sender to {}".format(ws_server))
+                    while True:
+                        if self._send_websockets:
+                            data = self.prepare_ws_data(self.triton)
+                            jsondata = json.dumps(data)
+                            log.debug("Sending data on websocket {}".format(ws_server))
+                            await websocket.send(jsondata)
+                        await asyncio.sleep(ws_sleep)
+            except OSError:
+                log.info("Websocket server is offline will retry later")
+                await asyncio.sleep(ws_connect_sleep)
+            except websockets.exceptions.ConnectionClosed:
+                log.info("Websocket closed. Will try to reconnect later")
+                await asyncio.sleep(ws_connect_sleep)
+
+    async def websocket_enable(self):
```

Comment:
Yes I guess I made it a web socket so that the ws server would only need to communicate over one protocol but it doesn't actually help since it's using a different port anyway